storage/remote/generic/generic.go

```diff
@@ -62,16 +69,15 @@ func (c *Client) Store(samples model.Samples) error {
 		}
 		req.Timeseries = append(req.Timeseries, ts)
 	}
-	client := NewGenericWriteClient(c.conn)
 	ctxt, _ := context.WithTimeout(context.Background(), c.timeout)
```

Comment:
Mh, the dial context is indeed confusing. 

Also I was phrasing things wrong/awkward previously. I'd just skip dialing with context now, but calling `Store()` should not instantiate a new background context. The API methods should rather get a context from the parent, i.e. `Store(ctx, samples)`.

My understanding is that there should be a logical context chain through your program. Otherwise they are really nothing but simple timeouts. It's not always convenient of course, especially when just introducing them. There's even a special TODO context for that (https://godoc.org/context#TODO).
But the caller of `Store` should ultimately be allowed to stop cancel any in-flight requests via the context.